src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -230,5 +230,12 @@ public static void ReadConversionFails()
 
             Assert.True(exceptionThrown);
         }
+
+        [Fact]
+        public static void ReadObject_PublicIndexer()
+        {
+            Indexer indexer = JsonSerializer.Parse<Indexer>(@"{""NonIndexerProp"":""Value""}");
+            Assert.Equal("Value", indexer.NonIndexerProp);
```

Comment:
nit: I would set this `_index` to some non-zero sentinel like -1 (within the `Indexer` class) and then verify that we are not clearing it on Parse.

So, add:
```C#
Assert.Equal(-1, indexer[0]);
```

Reply:
done